.github/workflows/release.yaml

```diff
@@ -1,72 +1,241 @@
+                  exit 1
+                fi                
+                ;;
+          
+              CLOSED)
+                echo "PR #$pull_request_number was closed, aborting the workflow."
+                exit 1
+                ;;
+          
+            esac
+          
+          done
+        continue-on-error: false
+
+      - name: Create and push new Git tag for release and pull latest changes
+        run: git tag ${{ inputs.irs-version }} && git push origin ${{ inputs.irs-version }} && git pull
 
       - name: Extract changelog text
         # See: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
         run: |
           EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
           echo "CHANGELOG<<$EOF" >> $GITHUB_ENV
-          sed -n -e '/## \[${{ inputs.new-irs-version }}\]/,/## \[/ p' CHANGELOG.md | head -n -1 | tail -n +2 >> $GITHUB_ENV
-          echo **Full Changelog**: ${{ github.server_url }}/${{ github.repository }}/compare/${{ inputs.previous-irs-version }}...${{ inputs.new-irs-version }} >> $GITHUB_ENV
+          sed -n -e '/## \[${{ inputs.irs-version }}\]/,/## \[/ p' ${{ env.CHANGELOG_PATH }} | head -n -1 | tail -n +2 >> $GITHUB_ENV
+          echo **Full Changelog**: ${{ github.server_url }}/${{ github.repository }}/compare\
+          /${{ steps.main-changelog-update.outputs.previous-irs-version }}...${{ inputs.irs-version }} >> $GITHUB_ENV
           echo "$EOF" >> "$GITHUB_ENV"
 
       - name: Create IRS release
         uses: softprops/action-gh-release@v2
         with:
           body: ${{ env.CHANGELOG }}
-          tag_name: ${{ inputs.new-irs-version }}
+          tag_name: ${{ inputs.irs-version }}
+
+  release-helm-chart:
+    name: "Release Helm chart"
+    needs:
+      - release
+    uses: ./.github/workflows/helm-chart-release.yaml
+    if: ${{ github.ref_name != 'main' }} # triggered automatically on main
+
+  build-irs:
+    name: "Build IRS"
+    needs:
+      - release
+    uses: ./.github/workflows/irs-build.yml
+    secrets: inherit
+    if: ${{ github.ref_name != 'main' }} # triggered automatically on main
+
+  trivy-image-scan:
+    name: "Scan image in local registry with Trivy"
+    needs:
+      - release
+    uses: ./.github/workflows/trivy-image-scan.yml
+    secrets: inherit
+    if: ${{ github.ref_name != 'main' }} # triggered automatically on main
 
-  publish-to-swaggerhub:
-    name: "Publish OpenAPI spec to Swaggerhub"
-    permissions:
-      contents: read
+  validate-openapi-definition:
     needs:
       - release
-    uses: ./.github/workflows/publish-swagger-hub.yml
-    with:
-      version: ${{ inputs.new-irs-version }}
-    secrets: inherit
\ No newline at end of file
+    name: "Validate OpenAPI definition"
+    uses: ./.github/workflows/swagger-editor-validate.yml
+    secrets: inherit
+    if: ${{ github.ref_name != 'main' }} # triggered automatically on main
```

Comment:
After the tag is created, the workflows should be run based on the tag, not the main branch. This is especially important for the docker image build step where the image is tagged based on the GitHub ref and the trivy image scan which should check the image build by the image build step

Reply:
Should now be resolved